src/main/java/com/sigmaukraine/messenger/controller/ChatController.java

```diff
@@ -0,0 +1,137 @@
+
+        return "chats/add";
+    }
+
+    @RequestMapping(value = "/add", method = RequestMethod.POST)
+    @PreAuthorize("isAuthenticated()")
+    public String addChat(@ModelAttribute("chat") Chat chat, BindingResult bindingResult, @PathVariable Integer subjectId) {
+        this.chatValidator.validate(chat, bindingResult);
+
+        //todo: fix base them
+        //todo: unique base
+
+        if (!chatRepository.isUnique(chat.getName(), subjectId)) {
+            bindingResult.rejectValue("name", "invalid.name", "Chat with this name already exist");
+            return "chats/add";
+        }
+        if (bindingResult.hasErrors()) {
+            return "chats/add";
+        }
+
+        User userDetails = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        com.sigmaukraine.messenger.domain.User user = userRepository.getUserByLogin(userDetails.getUsername());
+        chat.setCreatedBy(user.getId());
+        chat.setSubjectId(subjectId);
+        this.chatRepository.addChat(chat);
+        return "redirect:/subjects/"+ subjectId +"/chats";
+    }
+
+    @RequestMapping(value = "/edit/{chatId}", method = RequestMethod.GET)
+    @PreAuthorize("isAuthenticated()")
+    public String editChat(@PathVariable Integer subjectId, @PathVariable Integer chatId, Model model) {
+        if (!canEdit(chatId)) {
+            return "redirect:/subjects/"+ subjectId +"/chats";
+        }
+
+        Chat chat = chatRepository.getChatById(chatId);
+        model.addAttribute("chat", chat);
+        return "chats/edit";
+    }
+
+    @RequestMapping(value = "/edit/{chatId}", method = RequestMethod.POST)
+    @PreAuthorize("isAuthenticated()")
+    public String editChat(@PathVariable Integer subjectId, @PathVariable Integer chatId, @ModelAttribute("chat") Chat chat, BindingResult bindingResult) {
+        if (!canEdit(chatId)) {
+            return "redirect:/subjects/"+ subjectId +"/chats";
+        }
+
+        this.chatValidator.validate(chat, bindingResult);
+
+        if (bindingResult.hasErrors()) {
+            return "chats/edit";
+        }
+
+        if (!chatRepository.isUnique(chat.getName(), chatId)) {
+            bindingResult.rejectValue("name", "invalid.name", "Chat with this name already exist");
+            return "chats/add";
+        }
+        this.chatRepository.editChat(chatId, chat);
+        return "redirect:/subjects/"+ subjectId + "/chats";
+    }
+
+    @RequestMapping(value = "/remove/{chatId}", method = RequestMethod.GET)
+    @PreAuthorize("isAuthenticated()")
+    public String removeChat(@PathVariable Integer subjectId, @PathVariable Integer chatId){
+        if (canEdit(chatId)) {
+            this.chatRepository.removeChat(chatId);
+        }
+        return "redirect:/subjects/"+ subjectId + "/chats";
+    }
+
+    private boolean canEdit (Integer chatId) {
+        User userDetails = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
i think this check can be replaced with `@PreAuthorize("hasAuthority('admin')")` on a method you use this check in